examples/todoapp/test/feature_todo_list/blocs/todo_list_test.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:todoapp/feature_todo_list/blocs/todo_list_bloc.dart';
+
+import '../../Stubs.dart';
+import 'todo_list_test.mocks.dart';
+
+@GenerateMocks([
+  TodoListService,
+  CoordinatorBlocType,
+  CoordinatorEvents,
+  CoordinatorStates,
+])
+void main() {
+  late TodoListService _todoListService;
+  late CoordinatorBlocType _coordinatorBloc;
+  late CoordinatorEvents _coordinatorBlocEvents;
+  late CoordinatorStates _coordinatorBlocStates;
+
+  void _defineWhen() {
+    when(_todoListService.fetchTodoList())
+        .thenAnswer((_) => Future.value(Stubs.todoList));
+
+    when(_todoListService.filterTodos([], TodosFilterModel.all))
+        .thenAnswer((_) => []);
+
+    when(_todoListService.filterTodos(Stubs.todoList, TodosFilterModel.all))
+        .thenAnswer((_) => Stubs.todoList);
+
+    when(_todoListService.filterTodos(
+            Stubs.todoList, TodosFilterModel.completed))
+        .thenAnswer(
+            (_) => Stubs.todoList.where((todo) => todo.completed).toList());
+
+    when(_todoListService.filterTodos(
+            Stubs.todoList, TodosFilterModel.incomplete))
+        .thenAnswer(
+            (_) => Stubs.todoList.where((todo) => !todo.completed).toList());
+
+    when(_coordinatorBlocStates.onTodoAddedOrUpdated).thenAnswer(
+        (_) => Stream.value(Result.success(Stubs.todoUncompletedUpdated)));
+
+    when(_coordinatorBlocStates.onTodoDeleted)
+        .thenAnswer((_) => Stream.value(Result.success(Stubs.todoUncompleted)));
+
+    when(_coordinatorBlocStates.onTodoListChanged)
+        .thenAnswer((_) => Stream.value(Result.success(Stubs.todoListEmpty)));
+  }
+
+  TodoListBloc todoListBloc() => TodoListBloc(
+        _todoListService,
+        _coordinatorBloc,
+      );
+  setUp(() {
+    _todoListService = MockTodoListService();
+    _coordinatorBloc = MockCoordinatorBlocType();
+    _coordinatorBlocStates = MockCoordinatorStates();
+    _coordinatorBlocEvents = MockCoordinatorEvents();
+
+    when(_coordinatorBloc.states).thenReturn(_coordinatorBlocStates);
+    when(_coordinatorBloc.events).thenReturn(_coordinatorBlocEvents);
+  });
+
+  group('test todo_list_bloc_dart state todoList', () {
+    rxBlocTest<TodoListBlocType, Result<List<TodoModel>>>(
+        'test todo_list_bloc_dart state todoList',
+        build: () async {
+          _defineWhen();
+          return todoListBloc();
+        },
+        act: (bloc) async {},
+        state: (bloc) => bloc.states.todoList,
+        expect: [
+          Result.success([]),
+          Result.loading(),
+          Result.success(Stubs.todoList)
+        ]);
+  });
+
+  group('test todo_list_bloc_dart state filter', () {
+    rxBlocTest<TodoListBlocType, TodosFilterModel>(
+        'test todo_list_bloc_dart state filter',
+        build: () async {
+          _defineWhen();
+          return todoListBloc();
+        },
+        act: (bloc) async {
+          bloc.events.applyFilter(TodosFilterModel.all);
+          bloc.states.filter;
+        },
```

Comment:
You can `group` similar tests together inside a group's body. This way you don't have to add a group for each test separately.

```dart
group('name of the unifying group',(){
  test1();
  test2();
  rxBlocTest();
  // ...
});
```